src/renderer/port-forward/port-forward.store.ts

```diff
@@ -94,82 +104,248 @@ interface PortForwardResult {
   port: number;
 }
 
-interface PortForwardsResult {
-  portForwards: ForwardedPort[];
+function portForwardsEqual(portForward: ForwardedPort) {
+  return (pf: ForwardedPort) => (
+    pf.kind == portForward.kind &&
+    pf.name == portForward.name &&
+    pf.namespace == portForward.namespace &&
+    pf.port == portForward.port
+  );
 }
 
-export async function addPortForward(portForward: ForwardedPort): Promise<number> {
-  const { port, forwardPort } = portForward;
+function findPortForward(portForward: ForwardedPort) {
+  return portForwardStore.portForwards.find(portForwardsEqual(portForward));
+
+}
+
+const setPortForward = action((portForward: ForwardedPort) => {
```

Comment:
Is there a reason why those are not direct methods of the store class?

Reply:
I guess not anymore, good eye. I tried to keep the store isolated (somewhat) from the functions handling the ipc calls (`apiBase.post`, etc) because the store was just shadowing the `PortForward.portForwards` array managed in the main process. (you didn't need the store to create, or get, etc, port-forwards). But now the store is the "source of truth"

I could refactor in a followup PR.